### PR TITLE
feat(rule): add jsonschema-based options schema compiler

### PR DESCRIPTION
**Pull request by swwind**
## Summary

Step 1 of #1216: adds a JSON Schema-based options framework for rules (`internal/rule/schema.go`).

- `CompileSchema` / `Validate` compile a rule's options schema (Draft 4) and validate `context.options`, filling in schema-declared defaults the way ajv's `useDefaults` works under ESLint's config — including through `$ref`.
- Cross-checked against a real ajv@6 instance (configured exactly like ESLint) with 56 generated fixture cases: defaulting edge cases (nested/tuple/`$ref` gaps) plus real-world rule schemas from eslint-core, typescript-eslint, and popular plugins.

## Related Links

Part of #1216

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -67,6 +67,7 @@ require (
 	github.com/mackerelio/go-osstat v0.2.7 // indirect
 	github.com/mattn/go-colorable v0.1.14 // indirect
 	github.com/mattn/go-isatty v0.0.20 // indirect
+	github.com/santhosh-tekuri/jsonschema/v6 v6.0.2 // indirect
```
[the remark this answers is not in the document]
Comment (swwind) [reply go.mod]: Good catch — fixed in 448a4d67: `github.com/santhosh-tekuri/jsonschema/v6` is imported directly by `internal/rule/schema.go`, so it's now a direct requirement (ran `go mod tidy`).

File: internal/rule/schema.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+// CompileSchema compiles a rule's options JSON Schema. By convention the
+// schema describes the ESLint-style options array as a tuple: a top-level
+// `{"type": "array", "items": [...]}` — or, for a rule that takes no options,
+// `{"type": "array", "maxItems": 0}` with `items` omitted (draft-04's own
+// metaschema requires a non-empty schemaArray, so `"items": []` is itself
+// invalid). See the <rule-name>.schema.json convention.
+//
+// rslint only supports Draft 4 — the draft under which a plain array `items`
+// means positional/tuple validation — so authored schema.json files must not
+// declare a newer `$schema`.
```
[the remark this answers is not in the document]
Comment (swwind) [reply internal/rule/schema.go]: Fair point — softened in 448a4d67. This was aspirational wording; `CompileSchema` doesn't actually validate the declared `$schema`, it only sets Draft 4 as the *default* when a schema omits `$schema`. An explicit `$schema` for a newer draft would silently compile under that draft's own (different) `items` semantics. Since all schema.json files are authored in-repo and hand-reviewed, I don't think it's worth adding enforcement for this yet — the comment now just documents it as our convention rather than a guarantee.

File: internal/rule/schema.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+			val[i] = normalizeNumbers(v2)
+		}
+		return val
+	case json.Number:
+		f, _ := val.Float64()
+		return f
+	default:
+		return v
+	}
+}
```
[the remark this answers is not in the document]
Comment (swwind) [reply internal/rule/schema.go:426]: The map/slice mutation is intentional, not an oversight — it mirrors ajv's own `useDefaults`, which also inserts a compound default by reference into the instance being validated rather than deep-copying it (see the trade-off documented in `TestValidateDefaultsShareStorageAcrossCalls` right below in schema_test.go). This is a foundation PR with no caller yet, so there's no concurrent-use pattern to protect against today; if/when `Validate` does get called concurrently on the same `*CompiledSchema`, deep-copying compound defaults in `applyDefaults` is the fix, but I'd rather not pay that allocation on every call until it's actually needed.

On the ignored `Float64()` error: `normalizeNumbers` only ever sees `json.Number` values produced by decoding a schema's own `default` literal with `UseNumber()`, which by construction is already a syntactically valid JSON number — so that error path is unreachable for these inputs. Not adding a check for it to avoid implying a failure mode that can't occur here.

File: internal/rule/schema_test.go [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+	first.([]any)[0].(map[string]any)["config"].(map[string]any)["mutated"] = true
+
+	second, err := schema.Validate([]any{map[string]any{}})
+	if err != nil {
+		t.Fatalf("expected valid options, got error: %v", err)
+	}
+	secondConfig := second.([]any)[0].(map[string]any)["config"].(map[string]any)
+	if mutated, ok := secondConfig["mutated"]; !ok || mutated != true {
+		t.Errorf("expected second call's default to share storage with the first call's mutation, got %#v", secondConfig)
+	}
```
[the remark this answers is not in the document]
Comment (swwind) [reply internal/rule/schema_test.go:298]: This test is intentionally documenting the trade-off, not asserting a bug — see the comment directly above the assertion ("Deliberate trade-off: applyDefaults inserts the schema's compiled Default directly rather than deep-copying it..."). It's the same design choice as the `normalizeNumbers` mutation you flagged above: we're matching ajv's own `useDefaults` semantics, including its known footgun of sharing compound-default identity across validations of the same compiled schema. Locking this in with a test means if we ever do switch to deep-copying, we'll notice and update this test deliberately rather than being surprised by a behavior change.

**Review by fansenze — APPROVED** [review]
